backend/api/app/application/services/user_service.py

```diff
@@ -76,3 +107,152 @@ def verifySSID(self, ssid: str, ip_address: str):
             return True
         else:
             return False
+        
+    def logout_user(self, ssid: str, ip_address: str):
+            try:
+                if not ssid:
+                    return {"message": "SSID is required", "status": 400}
+
+                try:
+                    ses_id = uuid.UUID(ssid)
+                except ValueError:
+                    return {"message": "Invalid SSID format", "status": 400}
+
+                session = SessionRepository.get_active_by_id(ses_id, ip_address)
+                if not session:
+                    return {"message": "Session not found", "status": 404}
+
+                session.logged_out = True
+                SessionRepository.update(session)
+
+                return {"message": "User logged out successfully", "status": 200}
+
+            except ValueError as ve:
+                return {"message": str(ve), "status": 400}
+
+            except Exception as ex:
+                return {"message": "An unexpected error occurred", "details": str(ex), "status": 500}
+            
+    def delete_user(self, user_id: uuid.UUID, password: str):
+        try:
+            user = User.query.get(user_id)
+            if not user:
+                return {"message": "User not found", "status": 404}
+
+            if not user.check_password(password):
+                return {"message": "Incorrect password", "status": 401}
+
+            user.is_deleted = True  
+            UserRepository.updateUser(user)
+
+            return {"message": "User account deleted successfully", "status": 200}
+
+        except Exception as ex:
+            print(f"Error during delete operation: {str(ex)}")
+            return {"message": "An unexpected error occurred", "status": 500}
+        
+    def get_user_info(self, ssid: str, ip_address: str):
+        try:
+            ses_id = uuid.UUID(ssid)
+        except ValueError:
+            raise ValueError({"message": "Invalid session ID format.", "status": 400})
+        
+        session = SessionRepository.get_active_by_ssid(ses_id, ip_address)
+        if not session:
+            return {"message": "Session is not active or IP address mismatch.", "status": 401}
```

Comment:
We should not expose this type of data to user, we should just say you're not logged in. This message exposes our Authorization logic in a way.